Atividades/Em casa/deborah-gomes/clinicaVeterinaria.js

```diff
@@ -0,0 +1,103 @@
+function HistoricoMedico(consultas, vacinas, procedimentos) {
+    this.consultas = consultas || [];
+    this.vacinas = vacinas || [];
+    this.procedimentos = procedimentos || [];
+  }
+  
+  const historicoMedico = new HistoricoMedico(
+    ["9/7/2023", "13/1/2023", "20/7/2022"],
+    ["raiva", "v8", "v10"],
+    [{ tipo: "ultrassom", motivo: "dores" }]
+  );
+  
+  function Gato(nome, idade, cor, castrado, externo, historico) {
+    this.nome = nome;
+    this.idade = idade;
+    this.cor = cor;
+    this.castrado = castrado;
+    this.externo = externo;
+    this.historico = historico;
+  }
+  
+  Gato.prototype.miar = function () {
+    console.log("Miau!");
+  };
+  
+  Gato.prototype.vacinar = function (novaVacina) {
+    this.historico.vacinas.push(novaVacina);
+  };
+  
+  Gato.prototype.consultar = function (novaConsulta) {
+    this.historico.consultas.unshift(novaConsulta);
+  };
+  
+  const gatoVet = new Gato("Nina", 18, "frajola", true, false, historicoMedico);
+  
+  function Cachorro(nome, idade, cor, castrado, raca, historico) {
+    this.nome = nome;
+    this.idade = idade;
+    this.cor = cor;
+    this.castrado = castrado;
+    this.raca = raca;
+    this.historico = historico;
+  }
+  
+  Cachorro.prototype.latir = function () {
+    console.log("Au au!");
+  };
+  
+  Cachorro.prototype.vacinar = function (novaVacina) {
+    this.historico.vacinas.push(novaVacina);
+  };
+  
+  Cachorro.prototype.consultar = function (novaConsulta) {
+    this.historico.consultas.unshift(novaConsulta);
+  };
+  
+  const cachorroVet = new Cachorro("Zeus", 42, "preto", false, "labrador", historicoMedico);
+  
+  function Exotico(nome, idade, cor, especie, adulto, historico) {
+    this.nome = nome;
+    this.idade = idade;
+    this.cor = cor;
+    this.especie = especie;
+    this.adulto = adulto;
+    this.historico = historico;
+  }
+  
+  Exotico.prototype.vacinar = function (novaVacina) {
+    this.historico.vacinas.push(novaVacina);
+  };
+  
+  Exotico.prototype.consultar = function (novaConsulta) {
+    this.historico.consultas.unshift(novaConsulta);
+  };
+  
+  Exotico.prototype.getAdulto = function () {
```

Comment:
não teria que ter para as outras especies também?

```suggestion
  Exotico.prototype.getAdulto = function () {
   if (this.especie === "hamster") {
        return this.idade >= 2;
    }
    if (this.especie === "papagaio") {
        return this.idade >= 36;//considerando receber em meses
    }
    if (this.especie === "calopsita") {
        return this.idade >= 18;
    }
}
```

Reply:
Eu acho que ela meio que deu a opção de fazer só com 1 ou com os 3, não sei, não ficou muito claro, como não quis fazer vários ifs fiz de 1 só.